android/source/VideoLocker/src/org/edx/mobile/logger/Logger.java

```diff
@@ -2,10 +2,12 @@
 
 import android.content.Context;
 
+import java.io.Serializable;
+
 /**
  * Created by shahid on 22/1/15.
  */
-public class Logger {
+public class Logger implements Serializable {
```

Comment:
Here is a scenario where <code>Activity</code> is being stopped and trying to save state of  <code>PlayerFragment</code> hosted in it.  We retain <code>PlayerFragment</code> if activity is stopped and not being destroyed.

When <code>PlayerFragment</code> is being saved by its <code>Activity</code>, it tries to parcel (and in turn serialize) all its fields including <code>Logger</code> instance. 
I either have to mark the fields to be <code>transient</code> so that fragment will not save them or make them <code>Serializable</code> or let them be <code>static</code>.

@aleffert  I am not sure if there is better solution. Please suggest.

Here is the crash log for this scenario.

<code>
 java.lang.RuntimeException: Parcelable encountered IOException writing serializable object             at android.os.Parcel.writeSerializable(Parcel.java:1388)
            at android.os.Parcel.writeBundle(Parcel.java:663)
            at android.support.v4.app.FragmentState.writeToParcel(Fragment.java:133)
android.support.v4.app.FragmentManagerState.writeToParcel(FragmentManager.java:374)
            at android.os.Parcel.writeParcelable(Parcel.java:1357)
            at android.app.ActivityManagerProxy.activityStopped(ActivityManagerNative.java:2884)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
     Caused by: java.io.NotSerializableException: org.edx.mobile.logger.Logger
            at java.io.ObjectOutputStream.writeNewObject(ObjectOutputStream.java:1344)
            at java.io.ObjectOutputStream.writeFieldValues(ObjectOutputStream.java:959)
            at android.os.Parcel.writeSerializable(Parcel.java:1383)
            at android.os.Parcel.writeBundle(Parcel.java:663)
            at android.support.v4.app.FragmentState.writeToParcel(Fragment.java:133)
            at android.os.Parcel.writeTypedArray(Parcel.java:1191)
            at android.support.v4.app.FragmentManagerState.writeToParcel(FragmentManager.java:374)
            at android.os.Parcel.writeParcelable(Parcel.java:1357)
            at android.os.Parcel.writeBundle(Parcel.java:663)
            at android.app.ActivityManagerProxy.activityStopped(ActivityManagerNative.java:2884)
            at android.app.ActivityThread$StopInfo.run(ActivityThread.java:3261)
</code>